apps/condo/domains/notification/tasks/deliverMessage.js

```diff
@@ -46,132 +50,124 @@ async function _sendMessageByAdapter (transport, adapter, messageContext) {
-    if (!isEmpty(user)) {
-        transports.push(PUSH_TRANSPORT)
-    }
-
-    // At this point we return whatever non-empty sequence we've got
-    if (!isEmpty(transports)) return transports
-
-    // NOTE: none of requirements were met for message state, so we can't send anything anywhere actually.
-    throw new Error(`No appropriate transport found for notification id: ${id}`)
-}
-
 /**
  * Tries to deliver message via available transports depending on transport priorities
  * based on provided message data and available channels. If more prioritized channels fail message delivery,
- * tries to delived message through less prioritized fallback channels. Updates message status & meta in every case.
+ * tries to deliver message through less prioritized fallback channels. Updates message status & meta in every case.
  * @param messageId
  * @returns {Promise<string>}
  */
 async function deliverMessage (messageId) {
-    const { keystone } = await getSchemaCtx('Message')
-    const message = await Message.getOne(keystone, { id: messageId })
+    const { keystone: context } = await getSchemaCtx('Message')
+    const message = await Message.getOne(context, { id: messageId })
 
-    if (message.id !== messageId) throw new Error('get message by id has wrong result')
+    if (isEmpty(message)) throw new Error('get message by id has wrong result')
     // Skip messages that are already have been processed
     if (!MESSAGE_SENDING_STATUSES[message.status]) return `already-${message.status}`
 
-    const baseAttrs = {
-        dv: message.dv,
-        sender: message.sender,
-    }
+    const baseAttrs = { dv: message.dv, sender: message.sender }
+    const { error } = await checkMessageTypeInBlackList(context, message)
 
-    const { error } = await checkMessageTypeInBlackList(keystone, message)
     if (error) {
-        return await Message.update(keystone, message.id, {
+        const messageErrorData = {
             ...baseAttrs,
-            status: MESSAGE_ERROR_STATUS,
+            status: MESSAGE_BLACKLISTED_STATUS,
             processingMeta: {
                 dv: 1,
                 error,
             },
-        })
+        }
+
+        await Message.update(context, message.id, messageErrorData)
+
+        return MESSAGE_BLACKLISTED_STATUS
     }
 
-    const transports = await _choseMessageTransport(message)
-    const processingMeta = { dv: 1, transports, step: 'init' }
+    const processingMeta = { dv: 1, step: 'init' }
 
-    await Message.update(keystone, message.id, {
+    const messageInitData = {
         ...baseAttrs,
         status: MESSAGE_PROCESSING_STATUS,
         sentAt: null,
         deliveredAt: null,
         readAt: null,
         processingMeta,
-    })
+    }
+
+    await Message.update(context, message.id, messageInitData)
+
+    /** Let's extend DEFAULT_MESSAGE_DELIVERY_OPTIONS with MESSAGE_DELIVERY_OPTIONS[message.type] if available */
+    const { strategy, defaultTransports: transports } =
```

Comment:
You should extract this logic inside a function because you need to show another developer, that they can change this logic. For example, they can change choosing transport logic to get the transport from the user profile.